lib/mcp/app/resource_template.rb

```diff
@@ -53,57 +53,27 @@ def to_resource_template_hash
             mime_type: @mime_type,
             description: @description,
             handler: @handler,
-            variables: @variables
           }
         end
-
-        # Extract variables from a URI template
-        # e.g., "channels://{channel_id}" => ["channel_id"]
-        def extract_variables(uri_template)
-          variables = []
-          uri_template.scan(/\{([^}]+)\}/) do |match|
-            variables << match[0]&.to_sym
-          end
-          variables
-        end
-
-        # Creates a pattern for matching URIs against this template
-        def to_pattern
-          pattern_string = Regexp.escape(@uri_template).gsub(/\\\{[^}]+\\\}/) do |match|
-            "([^/]+)"
-          end
-          Regexp.new("^#{pattern_string}$")
-        end
-
-        # Extract variable values from a concrete URI based on the template
-        # e.g., template: "channels://{channel_id}", uri: "channels://123" => {"channel_id" => "123"}
-        def extract_variable_values(uri)
-          pattern = to_pattern
-          match = pattern.match(uri)
-          return {} unless match
-
-          result = {}
-          @variables.each_with_index do |var_name, index|
-            result[var_name] = match[index + 1]
-          end
-          result
-        end
       end
 
       def register_resource_template(uri_template, &block)
         builder = ResourceTemplateBuilder.new(uri_template)
         builder.instance_eval(&block)
         template_hash = builder.to_resource_template_hash
-        resource_templates[uri_template] = template_hash
+        template = Addressable::Template.new(uri_template)
+        resource_templates[template] = template_hash
```

Comment:
This is just nitpick territory but...

On a side note (though this issue was already in the code) - the fact that `@resource_templates` is a Hash is a bit misleading since it's actually never used like a Hash - it's just iterated element by element to find the match....

Which in my opinion would be better served by just being an `Array` and merging what was the key into the value